src/classes/PS_Request.cls

```diff
@@ -0,0 +1,293 @@
+*/
+public inherited sharing class PS_Request {
+
+    public enum ElevateEndpoint {
+        TOKENIZE,
+        PURCHASE,
+        COMMITMENT,
+        COMMITMENT_CANCEL
+    }
+
+    public enum ProductType {
+        ENGAGEMENT_HUB,
+        GIVING_PAGE,
+        CRM,
+        THIRD_PARTY_GIVING_PAGE
+    }
+
+    public static String ROLE_PAYMENT_TOKENS = 'Payments.Tokens';
+    public static String ROLE_PAYMENT_PURCHASE = 'Payments.Purchase';
+    public static String ROLE_PAYMENT_COMMITMENTS = 'Payments.Commitments';
+
+    public static String JWT_ISSUER_PARAM = 'iss';
+    public static String JWT_EXPIRES_AT_PARAM = 'exp';
+    public static String JWT_SUBJECT_PARAM = 'sub';
+    public static String JWT_SFDO_ID_PARAM = 'sfdo.id';
+    public static String JWT_PRODUCT_PARAM = 'sfdo.product';
+    public static String JWT_ROLES_PARAM = 'sfdo.roles';
+
+
+    /***
+    * @description Builds an HttpRequest
+    */
+    public class Builder {
+        /***
+        * @description Returns Elevate Payments Service configuration
+        */
+        private PS_IntegrationServiceConfig.Service configService {
+            get {
+                if (configService == null) {
+                    configService = new PS_IntegrationServiceConfig.Service();
+                }
+                return configService;
+            }
+            set;
+        }
+
+        private UTIL_Http.Method method;
+        private ElevateEndpoint endpoint;
+        private String jsonBody;
+        private String commitmentId;
+
+        /***
+        * @description Sets HttpRequest method
+        * @param method HttpRequest method
+        * @return Builder This request builder instance
+        */
+        public Builder withMethod(UTIL_Http.Method method) {
+            this.method = method;
+            return this;
+        }
+
+        /***
+        * @description Sets HttpRequest endpoint
+        * @param endpoint API endpoint
+        * @return Builder This request builder instance
+        */
+        public Builder withEndpoint(ElevateEndpoint endpoint) {
+            this.endpoint = endpoint;
+            return this;
+        }
+
+        /***
+        * @description Incorporates the Commitment Id into the endpoint where applicable
+        * @param commitmentId Recurring Donation Commitment Id
+        * @return Builder This request builder instance
+        */
+        public Builder withCommitmentId(String commitmentId) {
+            this.commitmentId = commitmentId;
+            return this;
+        }
+
+        /***
+        * @description Sets HttpRequest body
+        * @param jsonBody HttpRequest JSON body
+        * @return Builder This request builder instance
+        */
+        public Builder withBody(String jsonBody) {
+            this.jsonBody = jsonBody;
+            return this;
+        }
+
+        /***
+        * @description Builds a HttpRequest
+        * @return HttpRequest
+        */
+        public HttpRequest build() {
```

Comment:
The default callout timeout is 10 seconds. We are not setting timeout value explicitly on the request, but should we? If so, what would be optimal timeout for the commitment create, commitment cancel and purchase callout (it can be different for each or the same for all)?
@force2b , @andrewyu-salesforce , @ptewson-sfdo

Reply:
The change has been done to apply 20s timeout on the create commitment request. Other requests using the same `PS_Request.Builder()` do not have timeout value specified, meaning default 10s is applied by the core.